kotlinx-coroutines-core/common/src/flow/Builders.kt

```diff
@@ -315,3 +318,36 @@ private class ChannelFlowBuilder<T>(
     override fun toString(): String =
         "block[$block] -> ${super.toString()}"
 }
+
+private class CallbackFlowBuilder<T>(
+    private val block: suspend ProducerScope<T>.() -> Unit,
+    context: CoroutineContext = EmptyCoroutineContext,
+    capacity: Int = BUFFERED
+) : ChannelFlowBuilder<T>(block, context, capacity) {
+
+    private val collectCallback: suspend (ProducerScope<T>) -> Unit = {
```

Comment:
It looks like you can simply make `collectTo` open and override it here. There will be no need to override `broadcastImpl` and `produceImpl`.